modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,110 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	/// has_taur_sprite[icon_state]: Does this icon_state have ANY taur sprite?
+	var/static/list/has_taur_sprite = list()
+
+	/// has_taur_sprite_suit[icon_state]: has_taur_sprite but for /obj/item/clothng/suit
+	var/static/list/has_taur_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+/* NOTE FOR handle_taur_sprites(): This can be converted to handle things like digi sprites, vox, etc. I just didn't do it because I don't want to refactor the entire
+system for this one pr. ~niko */
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * Has seperate logic for suits, as they use a different set of DMIs as of now.
+ */
+/obj/item/clothing/proc/handle_taur_sprites()
+
+	if (istype(src, /obj/item/clothing/suit)) //override for suits, which use different dmis
+
+		if (mutant_variants & STYLE_TAUR)
+			return
+
+		if (has_taur_sprite_suit[icon_state]) // Very slightly speeds up the proc by letting things with no sprites instantly return
+			if (GLOB.naga_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_SNAKE
+			if (GLOB.horse_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_HOOF
+			if (GLOB.pawed_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_PAW
+			return
+
+		else if (has_taur_sprite_suit[icon_state] == FALSE)
+			return
+		// Code only goes here if has_taur_sprite_suit[] == nothing, AKA if init has never been ran
+		var/taur_sprite_suit = FALSE
+
+		if (GLOB.naga_taur_suit_sprites[icon_state])
+			has_taur_sprite_suit[icon_state] = TRUE
+			taur_sprite_suit = TRUE
+			mutant_variants |= STYLE_TAUR_SNAKE
+
+		if (GLOB.horse_taur_suit_sprites[icon_state])
+			has_taur_sprite_suit[icon_state] = TRUE //This block of code is checking the 3 DMI files for the icon state and setting flags/vars dynamically based on that
+			taur_sprite_suit = TRUE
+			mutant_variants |= STYLE_TAUR_HOOF
+
+		if (GLOB.pawed_taur_suit_sprites[icon_state])
+			has_taur_sprite_suit[icon_state] = TRUE
+			taur_sprite_suit = TRUE
+			mutant_variants |= STYLE_TAUR_PAW
+
+		if (!taur_sprite_suit) // If none of the 3 above if statements are true, it has no alt sprite
+			has_taur_sprite_suit[icon_state] = FALSE
+			return
```

Comment:
You should never need this, global init ought to be run before anything else

Reply:
What do you mean? This IS ran after global init, this is just how I'm checking to see if the icon_state has a taur sprite.

Reply:
Oh. Yeah, this is basically just kind of redundant right now, it's a caching system so we can skip the for loops if the clothing has no taur sprites. I'm debating removing it.

Reply:
Should have no reason to exist right?

Reply:
I honestly think there's no real reason to remove it, honestly, it's probably improving performance, even if VERY slight.

Reply:
when would it ever run? isn't it just dead code?

Reply:
> when would it ever run? isn't it just dead code?

I'm not sure what you mean, this runs on the first init of any clothing item, as long as the icon_state hasn't been used in this proc before. After that first init, the cacher is never ran again, but the cache is accessed on every successive init, for the purpose of immediately returning if there is no taur sprite.

It is a bit complicated for only a small performance increase, but again, I already wrote it, so

Reply:
What are you caching exactly? You already have a list of all valid icon states in those globals, what work needs to be cached?

Reply:
The other thing to note is even though it's already written, it requires upkeep just by being in the codebase. Most of the work that's done on code is done after it's been merged in